### DISCUSSION TITLE
Change GitHub username in CoPilot

### DISCUSSION THREAD

**Discussion by RoccoKing98** [community/community#125441] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I am using the VS Code Copilot extension and originally entered the wrong GitHub Username. How do I change the GitHub Username in the Copilot settings for VS Code?

**Comment by kazmatw** [comment]
To change the GitHub username associated with the VS Code Copilot extension, you can follow these steps:

1. Open VS Code Settings
Launch Visual Studio Code.
Open the Settings by either pressing Ctrl+, (comma) or by going to File > Preferences > Settings.
2. Navigate to Extensions Settings
In the Settings search bar, type "Copilot" to find all settings related to GitHub Copilot.
Look for any setting that might store the GitHub username or authentication settings.
3. Update the GitHub Credentials
If there's a direct option to change the username, update it accordingly.
If the username is tied to authentication, you might need to re-authenticate with the correct GitHub account. This can usually be done by logging out and then logging back in with the correct credentials. Look for a "Sign Out" or "Reauthenticate" button in the Copilot settings.
4. Reauthenticate Copilot
Sometimes, simply updating the username isn’t enough, especially if the extension uses OAuth tokens or similar methods for authentication. In this case, you might have to completely disconnect and reconnect GitHub Copilot.
To do this, use the command palette (Ctrl+Shift+P), type "Copilot: Sign Out", and hit Enter. Then, sign in again using the correct GitHub username.
5. Restart VS Code
After making the changes, restart VS Code to ensure that all settings are applied correctly and the new username is used.
If you cannot find the specific setting to change the username directly in the VS Code settings, check the GitHub Copilot documentation or GitHub settings on the web for more detailed guidance on managing linked applications and authentication sessions.

If you find this information useful, please consider selecting this as the answer to help others in the community identify effective solutions. Thank you!

**Reply by RoccoKing98** [reply]
Hello Kazmatw

Thank you very much for taking the time to write this solution. Unfortunately, there is no field in the extension settings that comes even close to being the authentication username. Do you have any idea where this information is stored? I have grep'ed my username under .vscode and .github, with no luck.

It could be that Copilot is taking my username from VSCode, but I am also not able to find this information stored anywhere.

For what its worth, I am running on an Ubuntu system.

Best Regards,

Edward

**Reply by RoccoKing98** [reply, marked as answer]
I finally found the solution. At the bottom left of the UI there is an "Accounts" icon 

![image](https://github.com/community/community/assets/140336921/a7fe3cb6-2071-4e47-96d7-8854973d4d04)

Clicking this, I was able to log out and then log back into the correct account.
